### DISCUSSION TITLE
Airflow content security policy

### DISCUSSION THREAD

**Discussion by superfive666** [apache/airflow#26011] [General]
Can Airflow configure the site to have content security policy enabled?

**Comment by potiuk** [comment]
No - but the best security practice if you care for it is to put Airflow behind proxy (for example nginx/apache etc. ) that can add it. this is typical enterprise practice - (similarly to http termination and authenitcation) - including OAuth etc. they are best done via enterprise proxy in front of applications behind.

**Comment by vandanthaker-ey** [comment]
> Can Airflow configure the site to have content security policy enabled?

@superfive666 did you figure this out? what CSP you applied?

**Reply by superfive666** [reply]
Yes, you have to apply the CSP at a proxy in front of the airflow application instead of doing it directly via any of the provided configuration.
However, you can still do it if you choose to have your own airflow template and inject it in your HTML header meta tag.

**Reply by vandanthaker-ey** [reply]
That I did. but do you have the directives which worked w/o using unsafe inline/eval? I tried 'default-src 'self' data:' , but it asks to use unsafe-inline
